roachpb/api.proto

```diff
@@ -741,6 +741,8 @@ message Header {
   // if set to a non-zero value, limits the total number of results for
   // Scan/ReverseScan requests in the batch.
   optional int64 max_scan_results = 8 [(gogoproto.nullable) = false];
+  // if set, all of the spans in the batch are distinct.
+  optional bool distinct_spans = 9 [(gogoproto.nullable) = false];
```

Comment:
The semantics of this seem fuzzy. Here it's talking about the batch, but at the point where it's set it looks at the intents for the whole transaction. That's still safe because the flag only matters when the transaction is a single batch (and saying "not distinct" is logically a safer option anyway), but I'd still prefer to be more precise about what this field means if we're going to introduce it on every request.